plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java

```diff
@@ -245,6 +228,40 @@ public MachineSource saveToSnapshot() throws MachineException {
         }
     }
 
+    /**
+     * Creates snapshot of workspace and push it to remote docker registry.
+     * This method has protected access because it can be extended to improve flexibility.
+     *
+     * @param image
+     *         snapshot image name (may include namespace)
+     * @param registry
+     *         registry to which snapshot image should be pushed
+     * @return Docker machine source
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    protected DockerMachineSource createAndPushSnapshotToRemoteRegistry(String image, String registry) throws IOException,
```

Comment:
It looks like this method is designed to be used in couple with [this](https://github.com/codenvy/codenvy/pull/629/files#diff-2264523513ff681f58e2aee07cd710bfR39) solution am i right ?
If so can you please answer my [question](https://github.com/codenvy/codenvy/pull/629/files#r76377299).